src/main/java/org/eclipse/tractusx/managedidentitywallets/config/MIWSettings.java

```diff
@@ -33,11 +36,128 @@
  * The type Miw settings.
  */
 @ConfigurationProperties(prefix = "miw")
-public record MIWSettings(String host, String encryptionKey, String authorityWalletBpn, String authorityWalletDid,
-                          String authorityWalletName,
-                          List<URI> vcContexts, List<URI> summaryVcContexts,
-                          @DateTimeFormat(pattern = "dd-MM-yyyy") Date vcExpiryDate,
-                          Set<String> supportedFrameworkVCTypes,
-                          boolean enforceHttps, String contractTemplatesUrl,
-                          List<URI> didDocumentContextUrls) {
+public class MIWSettings {
```

Comment:
Yeah the reason behind why we did that is because we had a problem in the converter, from string to SecretKey and we thought it maybe because the record is immutable. but now we managed it to work as well with record.
fixed in this commit [b5b3bd75](https://github.com/Cofinity-X/managed-identity-wallet/pull/1/commits/b5b3bd75220c77066a6c98665b9e45b09a8f1e3f)